include/vcmi/FactionMember.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * FactionMember.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#pragma once
+
+#include "Entity.h"
+
+VCMI_LIB_NAMESPACE_BEGIN
+
+class BonusList;
+
+namespace PrimarySkill
+{
+    enum PrimarySkill : int8_t;
+}
+
+class DLL_LINKAGE AFactionMember: public IConstBonusProvider, public INativeTerrainProvider
+{
+public:
+	/**
+	 Returns native terrain considering some terrain bonuses.
+	*/
+	virtual Identifier<ETerrainId> getNativeTerrain() const;
+	/**
+	 Returns magic resistance considering some bonuses.
+	*/
+	virtual int32_t magicResistance() const;
+	/**
+	 Returns minimal damage of creature or (when implemented) hero.
+	*/
+	virtual int getMinDamage(bool ranged) const;
+	/**
+	 Returns maximal damage of creature or (when implemented) hero.
+	*/
+	virtual int getMaxDamage(bool ranged) const;
+	/**
+	 Returns attack of creature or hero.
+	*/
+	virtual int getAttack(bool ranged) const;
+	/**
+	 Returns defence of creature or hero.
+	*/
+	virtual int getDefense(bool ranged) const;
+	/**
+	 Returns primskill of creature or hero.
+	*/
+	int getPrimSkillLevel(PrimarySkill::PrimarySkill id) const;
+	/**
+	 Returns morale or luck of creature or hero.
+	*/
+	int MoraleVal() const; //range [-3, +3]
+	int LuckVal() const; //range [-3, +3]
```

Comment:
Move "range [-3, +3]" comments into documentation comment above?

Also, perhaps give each method separate comment?

Reply:
It will be almost duplicate of comment from other val. I can, but do not see a reason. BTW, maybe add +5/-5 range setting in another PR?

Reply:
done

Reply:
> almost duplicate of comment from other val. 

True. Although not sure whether IDE's generate method tooltips (for autocomplete / hovering) for such shared comments. They definitely can show hints for individual comments though.

> maybe add +5/-5 range setting in another PR?

Why? H3 range is -3 ... +3.
If for purposes like modding - check game settings. This is where luck/morale chances are defined so we can try to connect these two - so if game settings define 5 options for positive morale we can return same range here.

Reply:
Range should be configurable btw, I am not sure if you setting made this possible. Ok, will check a setting.

Reply:
It is semi-configurable now, only within -3...+3 range. Negative luck is disabled in based game via 0 range.

There is one issue with range that needs to be resolved somehow - UI.
H3 has def file with images for -3, -2 ... +2 +3 luck/morale.
Changing range to higher range than H3 requires some handling in UI too

Reply:
Just show +3/-3 image. Anyway, it is not in scope of this PR, firstly we need to finish moving bonuses around.